### PR TITLE
Feature/tuff 37 events

### PR DESCRIPTION
**Pull request by ianballard**

### CODE REVIEW

File: test/unit_tests/tokenMaturity.js [hunk 1]
```diff
@@ -101,4 +103,141 @@ describe('TokenMaturity', function () {
...
+
+        expect(isMatured).to.equal(true, "should have reached maturity");
+    }
+
+    async function assertStartingTotalSupply() {
+        const startingTotalSupply = parseFloat(await tuffTokenDiamond.totalSupply());
+
+        const supplyNoDecimals = 1000000000;
+
+        expect(startingTotalSupply).to.equal(supplyNoDecimals * 10 ** 9, "incorrect total supply");
```
Comment (unibeck) [inline test/unit_tests/tokenMaturity.js:123]: Can you please use ethers helper functions to convert Wei into eth? I believe it is called parseeth or formateth. I should be using it else where in the test suite already

File: test/unit_tests/tokenMaturity.js [hunk 2]
```diff
@@ -101,4 +103,141 @@ describe('TokenMaturity', function () {
...
+
+        await expect(tuffTokenDiamond.onTokenMaturity())
+            .to.emit(tuffTokenDiamond, "TokenMatured")
+            .withArgs(new BN(contractStartingEthBalance.toString()), new BN(startingTotalSupply.toString()));
+
+        isLiquidated = await tuffTokenDiamond.getIsTreasuryLiquidated();
+
+        expect(isLiquidated).to.equal(true, "should have been liquidated");
+
+        const desiredAmountLeftOver = 100000 * 10 ** 9;
```
Comment (unibeck) [inline test/unit_tests/tokenMaturity.js:218]: Same as https://github.com/BuffChain/tufftoken/pull/18#discussion_r788273014

**Review by unibeck — APPROVED** [review]
Minor nitpick, but otherwise lgtm

File: test/unit_tests/tokenMaturity.js (hunk 1 above)
Comment (ianballard) [reply test/unit_tests/tokenMaturity.js:123]: this isnt converting wei to eth; its tuff

Comment (ianballard) [reply test/unit_tests/tokenMaturity.js]: updated magic decimal number

**Pull request merged**